lib/theme_check/json_printer.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+require 'json'
+
+module ThemeCheck
+  class JsonPrinter
+    def print(offenses)
+      json = offenses_by_path(offenses)
+      puts JSON.dump(json)
+    end
+
+    def offenses_by_path(offenses)
+      offenses
+        .map(&:to_h)
+        .group_by { |offense| offense[:path] }
+        .map do |(path, path_offenses)|
+          {
+            path: path,
+            messages: path_offenses.map { |offense| offense.filter { |k, _v| k != :path } },
```

Comment:
Should this keep be named `offenses` instead? It's a bit more than a message.
```suggestion
            offenses: path_offenses.map { |offense| offense.filter { |k, _v| k != :path } },
```

Reply:
Yes. That's an artefact of me ~copying~ being greatly influenced by the JSON format ESLint uses.